### PR TITLE
[SYCL][E2E] Add program scope host pipe test case for accelerator

### PR DESCRIPTION
**Pull request by cdai2**
Add the program scope host pipe test cases.

### CODE REVIEW

File: sycl/test-e2e/Basic/fpga_tests/hostpipe_base.cpp [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+    // Launch and Collect Test
+    std::cout << "Running Launch and Collect Test" << std::endl;
+    std::fill(out.begin(), out.end(), 0);
+    LaunchCollectTest(q, in.data(), out.data(), kPipeMinCapacity, 3);
+// std::cout << "Checking Launch/Collect\n";
+// for (int i = 0; i < count; i++) 
+  // std::cout << "in[" << i << "] = " << in[i] << "\n";
+// for (int i = 0; i < count; i++) 
+  // std::cout << "out[" << i << "] = " << out[i] << "\n";
```
Comment (rho180) [inline sycl/test-e2e/Basic/fpga_tests/hostpipe_base.cpp:118]: This commented code can be removed too.

**Comment by cdai2** [comment]
ping @intel/llvm-reviewers-runtime

**Comment by steffenlarsen** [comment]
1. Does this rely on https://github.com/intel/llvm/pull/9060? If so, it would probably be best to include the tests in that PR.
2. Is this expected to work on the FPGA emulator out-of-the-box?

**Comment by cdai2** [comment]
> 1. Does this rely on [[SYCL] Set compiled program for hostpipe #9060](https://github.com/intel/llvm/pull/9060)? If so, it would probably be best to include the tests in that PR.

Yes, these two cases rely on https://github.com/intel/llvm/pull/9060.  I will merge these two test cases in https://github.com/intel/llvm/pull/9060. 

> 2. Is this expected to work on the FPGA emulator out-of-the-box?

Yes.

**Pull request closed without merging**